src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -126,6 +145,36 @@ public final void onRunFinish(boolean succeeded, BatchSinkContext context) {
     }
   }
 
+  private void recordMetric(BatchSinkContext context, Job queryJob) throws Exception {
+    String operation = baseConfiguration.get(BigQueryConstants.CONFIG_OPERATION);
+    String metricName = String.format("records.%s", operation.toLowerCase());
+    long totalRows = getTotalRows(queryJob);
+    LOG.info(String.format("Total row count is %d" , totalRows));
+    //work around since StageMetrics count() only takes int as of now
+    final int cap = 10000; // so the loop will not cause significant delays
+    long count = totalRows / Integer.MAX_VALUE;
+    if (count > cap) {
+      LOG.warn("Record count too high! Metrics may not be updated correctly");
+    }
+    count = count < cap ? count : cap;
+    for (int i = 0; i <= count && totalRows > 0; i++) {
+      int rowCount = totalRows < Integer.MAX_VALUE ? (int) totalRows : Integer.MAX_VALUE;
+      context.getMetrics().count(metricName, rowCount);
+      totalRows -= rowCount;
+    }
+  }
+
+  private long getTotalRows(Job queryJob) throws Exception {
+    final JobConfiguration.Type type = queryJob.getConfiguration().getType();
+    LOG.info(String.format("Job configuration type is %s" , type));
```

Comment:
shouldn't log this, as it will always be dependent on the configured operation type.

Reply:
Removed.